### PR TITLE
Features/550 issubdtype

### PR DESCRIPTION
**Pull request by lenablind**
## Description

Implementation of function issubdtype based on [numpy analogue](https://numpy.org/doc/stable/reference/generated/numpy.issubdtype.html).
 

Issue/s resolved: #550

## Changes proposed:
Implementation of
- function issubdtype 
- corresponding tests

## Type of change
- New feature (non-breaking change which adds functionality)

## Due Diligence

- [x] All split configurations tested (not necessary)
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by mtar** [comment]
GPU cluster tests are currently disabled on this Pull Request.

**Comment by mtar** [comment]
GPU cluster tests are currently disabled on this Pull Request.

**Comment by lenablind** [comment]
rerun tests

**Comment by mtar** [comment]
run test

**Comment by mtar** [comment]
run tests

File: heat/core/types.py [hunk 1]
```diff
@@ -539,6 +540,29 @@ def can_cast(from_, to, casting="intuitive"):
...
+    arg1 : type, str, ht.dtype
+        A description representing the type. It may be a a Python builtin type, string or an HeAT type already.
+    arg2 : type, str, ht.dtype
+        A description representing the type. It may be a a Python builtin type, string or an HeAT type already.
+
+    Returns
+    -------
+    out : bool
+        Result of the stated comparison
+    """
```
Comment (coquelin77) [inline heat/core/types.py:558]: all looks good. can you add an example or two?

**Review by coquelin77 — COMMENTED** [review]
with an example in the docs this is good to go

File: heat/core/types.py (hunk 1 above)
Comment (lenablind) [reply heat/core/types.py:625]: Glad to hear - Yes, for sure.

**Review by coquelin77 — APPROVED** [review]

**Pull request merged**